### PR TITLE
Comment deletion

### PR DESCRIPTION
**Pull request by local-girl**
Hi there, this doesn't seem to have the desired effect for me, can you take a look please?

### CODE REVIEW

**Comment by ChristianSi** [comment]
@local-girl You think it's a backend issue?

**Comment by ChristianSi** [comment]
@local-girl There is a problem with the PUT request, it looks like this:

{"content_type":"adhocracy_core.resources.comment.IComment","data":{"adhocracy_core.sheets.metadata.IMetadata":{}}}

So the "hidden": "true" attribute is missing.

**Comment by ChristianSi** [comment]
CC @xi who has also worked on the deletion frontend.

**Comment by local-girl** [comment]
this is my put request:
```
return this.adhHttp.put(path, {
            content_type: "adhocracy_core.resources.comment.IComment",
            data: {
                "adhocracy_core.sheets.metadata.IMetadata": {
                    hidden: true
                }
            }
        });
```

**Comment by xi** [comment]
I found the issue: Before we actually send the request, we sanitize the data (`exportContent()`). In that function, there is the following code:

```
// workaround, as normal users can not set `hidden` field
// FIXME: use more appropriate place, e.g. expose in meta api
if (sheetName === "adhocracy_core.sheets.metadata.IMetadata" && fieldName === "hidden") {
    delete sheet[fieldName];
}
```

I am not sure whether this is still needed. An alternative would be to use `adhHttp.postRaw()` which does not do the sanitation. @nidico what do you think?

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Comment/CommentDetail.html [hunk 1]
```diff
@@ -24,6 +24,10 @@
                         <i class="icon-ban-circle"></i>
                         {{ "TR__REPORT" | translate }}
                     </a>
+                    <a href="#" data-ng-click="delete()" data-ng-if="report">
```
Comment (nidico) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Comment/CommentDetail.html]: This should not be `ng-if="report"`, but a proper check whether we are allowed to delete.

**Comment by nidico** [comment]
Yeah, well, this is still needed in order to make edit work. I regret that we ever introduced that, knowing that delete would break!

The proper solution would be getting the information from some OPTIONS response and remove the ugly sanitation hack.

**Comment by ChristianSi** [comment]
@nidico:

> The proper solution would be getting the information from some OPTIONS response and remove the ugly sanitation hack.

I would assume that that information is used already to decide whether to show the delete button in the first place. No need to check it twice, so allowing the delete button to turn off the sanitation seems reasonable enough.

**Comment by local-girl** [comment]
All yours @nidico  !

**Comment by nidico** [comment]
Deleting comments works, but still requires a reload after deletion. Also, an error is thrown if another user replies to the deleted comment in the meantime.

Deleting proposals does not work yet.

**Comment by nidico** [comment]
Ready for review.

**Comment by MagdaN** [comment]
Works for me!

**Pull request merged**